cmd/disk-cache.go

```diff
@@ -765,9 +761,7 @@ func (c *cacheObjects) PutObject(ctx context.Context, bucket, object string, r *
 	_, err = io.Copy(mwriter, r)
 	pipeWriter.Close()
 	wPipe.Close()
-
 	if err != nil {
-		err = <-errorCh
```

Comment:
client is expecting to see S3 API error returned by g/w endpoint, not multi write failures while writing to cache. The errorCh is necessary

Reply:
the error is written to the pipe @poornas so you don't need an `errorCh` the `io.Copy` returns the error that is seen on the other end.

Reply:
[cache put error](https://github.com/minio/minio/pull/14862/files#diff-88a5a1d895efc0e1b440c98d052f26dbdfd2e8ba1f329d830b1706f88d36a587L759) should not be cascaded up to handler

Reply:
I refactored since we are not interested in `dcache.put` error we can simply just log it and never use that to send it back over Pipe()

Reply:
The point is this 

```
        // ignore errors writing to cache
        if !t.pipeClosed {
                _, cerr := t.cacheWriter.Write(p)
                if cerr != nil {
                        t.pipeClosed = true
                        t.cacheWriter.CloseWithError(cerr)
                }
        }
        return len(p), nil
}
```

The errors generated by the cached writer are actually already ignored @poornas so there is no situation cache put error will be sent upwards to the handler